.cicdtemplate/.codemagic/codemagic.yaml

```
@@ -0,0 +1,145 @@
+definitions:
+  configure_environment: &configure_environment
+    # This running machine can be changed depends on the billing plan: https://docs.codemagic.io/knowledge-codemagic/machine-type/
+    instance_type: mac_mini_m1
+    max_build_duration: 30
+    environment:
+      groups:
+        - firebase_credentials
+    cache:
+      cache_paths:
+        - $HOME/.gradle/caches
+  scripts:
+    - &set_up_google_services_files_from_environment_variables
+      name: Set up google-services.json on template-compose
+      script: |
+        mkdir -p template-compose/app/src/production
+        echo $GOOGLE_SERVICES_JSON > template-compose/app/src/production/google-services.json
+        mkdir -p template-compose/app/src/staging
+        echo $GOOGLE_SERVICES_JSON_STAGING > template-compose/app/src/staging/google-services.json
+    - &detekt_on_template_compose
+      name: Run detekt on template-compose
+      working_directory: ./template-compose
+      script: ./gradlew detekt
+    - &detekt_on_template_xml
+      name: Run detekt on template-xml
+      working_directory: ./template-xml
+      script: ./gradlew detekt
+    - &unit_test_on_template_compose
+      name: Run unit tests on template-compose
+      working_directory: ./template-compose
+      script: ./gradlew koverMergedReport
+    - &unit_test_on_template_xml
+      name: Run unit tests on template-xml
+      working_directory: ./template-xml
+      script: ./gradlew koverMergedReport
+  artifacts:
+    - &artifacts_template_compose template-compose/build/reports/kover/merged/
+    - &artifacts_template_xml template-xml/build/reports/kover/merged/
+    - &artifacts_staging_apk template-compose/app/build/outputs/apk/staging/debug/app-staging-debug.apk
+    - &artifacts_production_apk template-compose/app/build/outputs/apk/production/debug/app-production-debug.apk
```

same for Codemagic in the template, no `template-compose` or `template-xml`, only `app` at root 😄 Please apply for the rest in this file 🙏 

```suggestion
    - &set_up_google_services
      name: Set up google-services.json
      script: |
        mkdir -p app/src/production
        echo $GOOGLE_SERVICES_JSON > app/src/production/google-services.json
        mkdir -p app/src/staging
        echo $GOOGLE_SERVICES_JSON_STAGING > app/src/staging/google-services.json
    - &detekt
      name: Run detekt
      script: ./gradlew detekt
    - &unit_test
      name: Run unit tests
      script: ./gradlew koverMergedReport
  artifacts:
    - &artifacts_template_compose build/reports/kover/merged/
    - &artifacts_staging_apk app/build/outputs/apk/staging/debug/app-staging-debug.apk
    - &artifacts_production_apk app/build/outputs/apk/production/debug/app-production-debug.apk
```